modules/db/src/test/java/org/mule/module/db/integration/update/UpdateBulkParameterizedDebugInfoTestCase.java

```diff
@@ -0,0 +1,103 @@
+import static org.mule.module.db.internal.processor.DbDebugInfoUtils.SQL_TEXT_DEBUG_FIELD;
+import static org.mule.module.db.internal.processor.DbDebugInfoUtils.TYPE_DEBUG_FIELD;
+import static org.mule.tck.junit4.matcher.ObjectDebugInfoMatcher.objectLike;
+import org.mule.api.MuleEvent;
+import org.mule.api.debug.FieldDebugInfo;
+import org.mule.api.processor.MessageProcessor;
+import org.mule.construct.Flow;
+import org.mule.module.db.integration.AbstractDbIntegrationTestCase;
+import org.mule.module.db.integration.TestDbConfig;
+import org.mule.module.db.integration.model.AbstractTestDatabase;
+import org.mule.module.db.internal.domain.query.Query;
+import org.mule.module.db.internal.processor.AbstractDbMessageProcessor;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Test;
+import org.junit.runners.Parameterized;
+
+public class UpdateBulkParameterizedDebugInfoTestCase extends AbstractDbIntegrationTestCase
+{
+
+    public static final String PARAM1 = PARAM_DEBUG_FIELD_PREFIX + 1;
+    public static final String PARAM_SET1 = PARAM_SET_DEBUG_FIELD_PREFIX + 1;
+    public static final String PARAM_SET2 = PARAM_SET_DEBUG_FIELD_PREFIX + 2;
+
+    public UpdateBulkParameterizedDebugInfoTestCase(String dataSourceConfigResource, AbstractTestDatabase testDatabase)
+    {
+        super(dataSourceConfigResource, testDatabase);
+    }
+
+    @Parameterized.Parameters
+    public static List<Object[]> parameters()
+    {
+        return TestDbConfig.getDerbyResource();
+    }
+
+    @Override
+    protected String[] getFlowConfigurationResources()
+    {
+        return new String[] {"integration/update/update-bulk-config.xml"};
+    }
+
+    @Test
+    public void providesDebugInfo() throws Exception
+    {
+        Flow flowConstruct = (Flow) muleContext.getRegistry().lookupFlowConstruct("updateBulk");
+
+        List<MessageProcessor> messageProcessors = flowConstruct.getMessageProcessors();
+        AbstractDbMessageProcessor queryMessageProcessor = (AbstractDbMessageProcessor) messageProcessors.get(0);
+
+        List<String> planetNames = new ArrayList<>();
+        planetNames.add(EARTH.getName());
+        planetNames.add(MARS.getName());
+
+        final MuleEvent muleEvent = getTestEvent(planetNames);
+
+        final List<FieldDebugInfo> debugInfo = queryMessageProcessor.getDebugInfo(muleEvent);
+
+        assertThat(debugInfo, is(not(nullValue())));
+        assertThat(debugInfo.size(), equalTo(2));
+
+        List<FieldDebugInfo> queryFields = new ArrayList<>();
+        queryFields.add(createFieldDebugInfo(SQL_TEXT_DEBUG_FIELD, String.class, "update PLANET set NAME='Mercury' where NAME=?"));
+        queryFields.add(createFieldDebugInfo(TYPE_DEBUG_FIELD, String.class, UPDATE.toString()));
+        assertThat(debugInfo, hasItem(objectLike(QUERY_DEBUG_FIELD, Query.class, queryFields)));
+
+        final List<FieldDebugInfo> paramSets = new ArrayList<>();
```

Comment:
I think it will be clearer if this list is build in another method, like in the other tests below.